src/index.js

```diff
@@ -1 +1,75 @@
+/* eslint-disable no-console */
 'use strict';
+
+const http = require('http');
+const fs = require('fs');
+const multiparty = require('multiparty');
+const { getCompressionType } = require('./getCompressionType');
+
+const PORT = process.env.PORT || 3000;
+
+const server = http.createServer((req, res) => {
+  if (req.method === 'POST') {
+    const form = new multiparty.Form();
+
+    form.parse(req, (err, fields, files) => {
+      if (err) {
+        res.statusCode = 500;
+        res.end('Server Error');
+
+        return;
+      }
+
+      if (!files.file || !files.file[0]) {
+        res.statusCode = 400;
+        res.end('Bad Request: No file uploaded');
+
+        return;
+      }
+
+      const file = files.file[0];
+      const fileName = file.originalFilename;
+
+      const compressionMethod = fields.compression[0];
+      const compression = getCompressionType(compressionMethod);
+      const compressedFilePath
+        = `${__dirname}/${fileName}.${compressionMethod}`;
+
+      const readStream = fs.createReadStream(file.path);
+      const writeStream = fs.createWriteStream(compressedFilePath);
+
+      readStream.pipe(compression).pipe(writeStream);
```

Comment:
The code does not handle stream errors. It would be good to handle possible errors on read/write streams to ensure the application does not crash in case of a stream error.

Reply:
https://github.com/mate-academy/node_compression-app/pull/53/files#r1315967562
not fixed